packages/core/src/store/level-state-store.ts

```diff
@@ -36,7 +36,7 @@ export class LevelStateStore implements StateStore {
      * @param docStateHolder - Document instance
      */
     async save(docStateHolder: DocStateHolder): Promise<void> {
-        await this.#store.put(docStateHolder.id.baseID.toString(), DoctypeUtils.serializeState(docStateHolder.state))
+        await this.#store.put(docStateHolder.id.toString(), DoctypeUtils.serializeState(docStateHolder.state))
```

Comment:
why this change?  Don't we always want to store by baseID?  I'm not sure it makes sense to cache state at specific commitIds in the state store.

Maybe we should just throw if the doc is pinned to a specific CommitID when trying to store it into the state store?

Reply:
Storing states for specific CommitIDs in the state store might make sense when you are loading a commit of a document multiple times. This happens in 3ID and for schemas for example.

Reply:
But you could just store the current state of the doc in the state store, load it, then reset to the old version

Reply:
That's going to be quite slow if you are loading it multiple times.

Reply:
Ideally that's the case yes. I would guess that's not how it's implemented right now however.

Reply:
Surprised about this. `DocStateHolder#id` is defined as `DocID` type. `DocID#baseID` is just redundant here.

Reply:
> Ideally that's the case yes. I would guess that's not how it's implemented right now however.

Shouldn't the plan then be to move in the direction towards optimizing the implementation rather than attempting to optimize by storing more data? @oed

Reply:
> Surprised about this. `DocStateHolder#id` is defined as `DocID` type. `DocID#baseID` is just redundant here.

Ah right, sorry, my mistake.

How do we ensure that we aren't putting the state from a "readonly" document at was loaded at a specific CommitID into the state store as if it was the most recent version of that document?  Can we add an assert somewhere into this process that would catch that type of programming or user error?  It would be a pretty nasty bug if we ever accidentally overrode the state store with less up-to-date info for a given DocID than what was already there

Reply:
@stbrody I believe, this check could be made on type level. Right now all of RunningState, RunningStateLike, SnapshotState along with Doctype implement DocStateHolder interface. Maybe at the next iteration, we think on proper typing to avoid this kind of accident.